### PR TITLE
br: add additional volumes to backup/restore/warmup pods

### LINKED ISSUE #5387: ebs br: quickly timeout the ebs br operation
## Feature Request

**Is your feature request related to a problem? Please describe:**

**Describe the feature you'd like:**

If pod scheduling for br operation like backup/restore/warmup/br can't be finished in time, fail the whole BR quickly. And also , we need to add TTL support for backup init pod in case GC / PD scheduler can't be resumed.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### LINKED ISSUE #5404: ebs br: support mount extra volumes to started Pod
## Feature Request

**Is your feature request related to a problem? Please describe:**

**Describe the feature you'd like:**

As the title. Extra volume mount is needed to emit logs.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### LINKED ISSUE #5409: TTL on init pod for EBS backup
## Feature Request

**Is your feature request related to a problem? Please describe:**
The backup init pod is used to acquire the lease for disabling GC. And managed by the backup manager/controller in the operator. If the operator were to die (or reverted to an incompatible version - this happened for us), this will keep the gc disabled. Putting the cluster in a very unhealthy state. 

**Describe the feature you'd like:**
The init pod typically runs only for a minute, while the backup pod, triggers async snapshot creation. To the request here is to run the init pod with a TTL (of lets say 5 mins), after which it'll kill itself. If the backup pod, can check if init is still running, or hasn't completed. If it has completed, before trigger snapshot creation, then the backup pod can fail. 

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by WangLe1321**
### What problem does this PR solve?

Add additional volumes and volume mounts to backup/restore/warmup pods, then we can support users to mount `configMap` these pods.

Closes #5404 

Add deadline to backup-init job of 10 minutes, in case it blocks gc and pd schedule indefinitely.

Closes #5387
Closes #5409

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

I add `additionalVolumes` and `additionalVolumeMounts` to the `VolumeBackup`
```
    additionalVolumeMounts:
    - mountPath: /etc/foo
      name: foo
      readOnly: true
    additionalVolumes:
    - configMap:
        name: br-test
      name: foo
```

I see the volume and volume mount appears in the backup and backup-init pod

```
    volumeMounts:
    - mountPath: /etc/foo
      name: foo
      readOnly: true

  volumes:
  - configMap:
      defaultMode: 420
      name: br-test
    name: foo
```

for deadline of backup init job, I see the corresponding fields appearing in the job `activeDeadlineSeconds: 600`

and the job failed due to the deadline

```
status:
  conditions:
  - lastProbeTime: "2023-11-23T09:07:17Z"
    lastTransitionTime: "2023-11-23T09:07:17Z"
    message: Job was active longer than specified deadline
    reason: DeadlineExceeded
    status: "True"
    type: Failed
```

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Review by BornChanger — APPROVED** [review]

**Comment by nkg-** [comment]
> Add deadline to backup-init job of 10 minutes, in case it blocks gc and pd schedule indefinitely.

Thanks for supporting this. My 2c is to move to another PR. Since its a logically different change, than indicated in the title.

File: pkg/backup/backup/backup_manager.go [hunk 1]
```diff
@@ -846,6 +855,15 @@ func (bm *backupManager) makeBRBackupJob(backup *v1alpha1.Backup) (*batchv1.Job,
 		},
 	}
 
+	// for volume backup initializing job, we should set resource requirement empty
+	// avoid it consuming too much resource
+	if backup.Spec.Mode == v1alpha1.BackupModeVolumeSnapshot &&
+		backup.Spec.FederalVolumeBackupPhase == v1alpha1.FederalVolumeBackupInitialize {
+		bm.setBackupPodResourceRequirementsEmpty(&job.Spec.Template)
+		// for volume backup initializing job, set deadline of the job in case it blocks GC and pd schedule indefinitely
+		job.Spec.ActiveDeadlineSeconds = pointer.Int64(int64(volumeBackupInitJobMaxActiveSeconds))
```
Comment (wxiaomou) [inline pkg/backup/backup/backup_manager.go:864]: @WangLe1321 this gave me compile error `pkg/backup/backup/backup_manager.go:864:44: undefined: pointer.Int64` should we use pointer.Int64Ptr instead?

**Pull request merged**